lib/utils/index.js

```diff
@@ -33,6 +34,10 @@ utils.falsey = value => {
   return false;
 };
 
+utils.isOptions = val => _.isObject(val) && _.isObject(val.hash);
+
+utils.isUndefined = val => val == null || (utils.isOptions(val) && val.hash != null);
```

Comment:
Why this problem was not caught by the tests? :thinking:

Reply:
because unit tests don't have strings as testing data, only arrays
I've added 3 cases with string